### PR TITLE
Clang format

### PR DESCRIPTION
**Pull request by arrooney**
Users of the repo can now run `./install-hooks` to install the githooks permanently (unless the hooksPath config is changed. This will install/run clang-format with our style rules in place before committing.

### CODE REVIEW

**Review by Taylorr82 — APPROVED** [review]
If these changes are successful will you go on to do all of the submodules as well? The athena_software one really needs attention

**Review by ArashYzd — APPROVED** [review]
Does it only apply to this repo? not the submodules?
Do we need to run the command each time we make a new clone?
I also think we should mention this in addition to all the git stuff necessary for the submodules to the readme.

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,137 @@
+---
+Language:        Cpp
```
Comment (Taylorr82) [inline .clang-format:2]: Shouldn't this be C?

**Comment by Taylorr82** [comment]
I'm wondering if we should close this because of recent changes and push a new one faster

**Comment by Taylorr82** [comment]
Closing to redo with updated code

**Pull request closed without merging**